homeassistant/components/system_bridge/media_source.py

```diff
@@ -0,0 +1,209 @@
+                    )
+                )
+
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier="",
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type="",
+            title=self.name,
+            can_play=False,
+            can_expand=True,
+            children=children,
+            children_media_class=MEDIA_CLASS_DIRECTORY,
+        )
+
+
+def _build_base_url(
+    entry: ConfigEntry,
+) -> str:
+    """Build base url for System Bridge media."""
+    return f"http://{entry.data[CONF_HOST]}:{entry.data[CONF_PORT]}/api/media/file/data?apiKey={entry.data[CONF_API_KEY]}"
+
+
+def _build_root_paths(
+    entry: ConfigEntry,
+    media_directories: MediaDirectories,
+) -> BrowseMediaSource:
+    """Build base categories for System Bridge media."""
+    return BrowseMediaSource(
+        domain=DOMAIN,
+        identifier="",
+        media_class=MEDIA_CLASS_DIRECTORY,
+        media_content_type="",
+        title=entry.title,
+        can_play=False,
+        can_expand=True,
+        children=[
+            BrowseMediaSource(
+                domain=DOMAIN,
+                identifier=f"{entry.entry_id}~~{directory.key}",
+                media_class=MEDIA_CLASS_DIRECTORY,
+                media_content_type="",
+                title=f"{directory.key[:1].capitalize()}{directory.key[1:]}",
+                can_play=False,
+                can_expand=True,
+                children=[],
+                children_media_class=MEDIA_CLASS_DIRECTORY,
+            )
+            for directory in media_directories.directories
+        ],
+        children_media_class=MEDIA_CLASS_DIRECTORY,
+    )
+
+
+def _build_media_items(
+    entry: ConfigEntry,
+    media_files: MediaFiles,
+    path: str,
+    identifier: str,
+) -> BrowseMediaSource:
+    """Fetch requested files."""
+    return BrowseMediaSource(
+        domain=DOMAIN,
+        identifier=identifier,
+        media_class=MEDIA_CLASS_DIRECTORY,
+        media_content_type="",
+        title=f"{entry.title} - {path}",
+        can_play=False,
+        can_expand=True,
+        children=[
+            _build_media_item(identifier, file)
+            for file in media_files.files
+            if file.is_directory
+            or (
+                file.is_file
+                and file.mime_type is not None
+                and file.mime_type.startswith(MEDIA_MIME_TYPES)
+            )
+        ],
+    )
+
+
+def _build_media_item(
+    path: str,
+    media_file: MediaFile,
+) -> BrowseMediaSource:
+    """Build individual media item."""
+    ext = (
+        f"~~{media_file.mime_type}"
+        if media_file.is_file and media_file.mime_type is not None
+        else ""
+    )
+    return BrowseMediaSource(
+        domain=DOMAIN,
+        identifier=f"{path}/{media_file.name}{ext}",
+        media_class=MEDIA_CLASS_DIRECTORY
+        if media_file.is_directory or media_file.mime_type is None
```

Comment:
A ternary operator expression inside an instantiating multi-line call like this is really hard to read. Please rewrite it.